src/lib.rs

```diff
@@ -103,12 +103,133 @@ impl TryFrom<u8> for HashAlgorithm {
         match value {
             // Naturally, these need to match the enum declaration
             1 => Ok(Self::MurmurHash3),
-            2 => Ok(Self::Sha256),
+            2 => Ok(Self::Sha256l32),
+            3 => Ok(Self::Sha256),
             _ => Err(()),
         }
     }
 }
 
+trait CascadeIndexGenerator {
```

Comment:
Might be good to write a bit of documentation for this.

Reply:
Yeah, and as I was writing it up I realized it'd be more natural if `next_layer` set the range for all calls to `next_index` in that layer. So I made that change too.